site/src/jsMain/kotlin/com/google/youtube/pages/ChannelPage.kt

```diff
@@ -61,6 +63,14 @@ import org.jetbrains.compose.web.dom.Text
 
 @Composable
 fun ChannelPage(id: String, initialTab: ChannelTab?) {
+    val navigator = LocalNavigator.current
+    var selectedTab by remember { mutableStateOf(initialTab ?: ChannelTab.Home) }
+    
+    // Update selected tab when route changes
+    LaunchedEffect(initialTab) {
+        selectedTab = initialTab ?: ChannelTab.Home
+    }
```

Comment:
Let's introduce `initialTab` as a key to `selectedTab`'s `remember` over using `LaunchedEffect`.